### PR TITLE
unmaintained b* packages: update to use f-strings

### PR DESCRIPTION
**Pull request by alecbcs**
Update legacy .format() calls to f-strings.

### CODE REVIEW

File: var/spack/repos/builtin/packages/bamtools/package.py
Comment (haampie) [inline var/spack/repos/builtin/packages/bamtools/package.py]:
```suggestion
        args.append(self.define("CMAKE_INSTALL_RPATH", rpath))
```

File: var/spack/repos/builtin/packages/bazel/package.py
Comment (haampie) [inline var/spack/repos/builtin/packages/bazel/package.py]: ?

File: var/spack/repos/builtin/packages/bigdft-core/package.py
Comment (haampie) [inline var/spack/repos/builtin/packages/bigdft-core/package.py:84]: ^

File: var/spack/repos/builtin/packages/blast2go/package.py
Comment (haampie) [inline var/spack/repos/builtin/packages/blast2go/package.py]: combine?

File: var/spack/repos/builtin/packages/boxlib/package.py
Comment (haampie) [inline var/spack/repos/builtin/packages/boxlib/package.py]: also here, just use self.define(key, val)

File: var/spack/repos/builtin/packages/branson/package.py
Comment (haampie) [inline var/spack/repos/builtin/packages/branson/package.py]: and here

**Review by haampie — CHANGES_REQUESTED** [review]
Is this automated?

File: var/spack/repos/builtin/packages/bazel/package.py
Comment (alecbcs) [reply var/spack/repos/builtin/packages/bazel/package.py]: Good catch fixed.

File: var/spack/repos/builtin/packages/blast2go/package.py
Comment (alecbcs) [reply var/spack/repos/builtin/packages/blast2go/package.py]: Done!

File: var/spack/repos/builtin/packages/boxlib/package.py
Comment (alecbcs) [reply var/spack/repos/builtin/packages/boxlib/package.py]: Hadn't used `self.define()` before, but I really like it. We should make sure more folks know about this!

File: var/spack/repos/builtin/packages/bigdft-core/package.py
Comment (alecbcs) [reply var/spack/repos/builtin/packages/bigdft-core/package.py:84]: Are you suggesting to use `self.define()` here as well? How does that work for `--with` statements?

**Comment by alecbcs** [comment]
Partially yes, I ran `pyupgrade` (suggestion from @adamjstewart), and `flynt`, across the set of files and then manually fixed up a couple of the edge cases the tools didn't do so well with.

**Comment by adamjstewart** [comment]
If this is primarily automated I'm opposed to doing it letter by letter. We should do it all in one commit so we can ignore it in `.git-blame-ignore-revs`.

**Comment by alecbcs** [comment]
I wouldn't be opposed to running the full repository in a completely automated run as a single PR, but we're going to need to merge that PR quickly or else we'll be continuously fighting merge conflicts. 

As is I'm still doing a fair amount of the rewrites by hand and so I'd figured bundling unmaintained packages by letter might be the sweet spot between bulk edits and not loosing my mind working through the whole repository at once only to have to figure out merge conflicts soon after.

**Comment by adamjstewart** [comment]
I guess my preference would be a single PR that adds pyupgrade and/or flynt as a style test and includes autoformat-only changes, and then follow-up PRs as needed to manually update packages. But I'll let others decide.

**Comment by alecbcs** [comment]
I think that makes a lot of sense. Are you opposed to closing these two PRs prior to our full repository run?
(Since these are a mix of automation and hand written fixes.)

**Comment by adamjstewart** [comment]
I'm fine with that.

File: var/spack/repos/builtin/packages/bigdft-atlab/package.py
Comment (haampie) [inline var/spack/repos/builtin/packages/bigdft-atlab/package.py:51]:
```suggestion
            f"--with-futile-incs={spec['bigdft-futile'].headers.include_flags}/futile",
```

File: var/spack/repos/builtin/packages/bigdft-core/package.py
Comment (haampie) [reply var/spack/repos/builtin/packages/bigdft-core/package.py:84]: No, I meant it was missing an f-string, dunno if oversight or intentional.

File: var/spack/repos/builtin/packages/bigdft-libabinit/package.py
Comment (haampie) [inline var/spack/repos/builtin/packages/bigdft-libabinit/package.py:52]: Also missing f-string? (Not entirely sure what this + "f90" is supposed to do.

**Pull request closed without merging**
